PR: Use a better icon when no matches are found (Find/Replace widget)

ccordoba12 opened the pull request:
## Description of Changes

- The current icon (the warning one) did not represent correctly the lack of matches after searching for text.

    **Before**

    ![imagen](https://user-images.githubusercontent.com/365293/209702827-329ce12a-f1f0-40a1-a0dc-a554df0ca75c.png)

    **After**

    ![image](https://user-images.githubusercontent.com/365293/209750869-01517391-bfba-4971-8d19-7f1e2c1c519b.png)


- This is a follow-up of #20049.

### Issue(s) Resolved

<!--- List the issue(s) below, in the form "Fixes #1234"; one per line --->

Fixes #20255.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct:

<!--- Thanks for your help making Spyder better for everyone! --->

dalthviz reviewed (commented):
Thanks @ccordoba12 ! I think the new icon looks good. Also while checking this I got a traceback related to the findreplace code and the split editor/new window functionality:

```
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2265, in focus_changed
    self.refresh()
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2425, in refresh
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2219, in current_changed
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2265, in focus_changed
    self.refresh()
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2425, in refresh
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
Traceback (most recent call last):
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\plugins\editor\widgets\editor.py", line 2219, in current_changed
    self.find_widget.set_editor(editor, refresh=False)
  File "e:\acer\documentos\spyder\spyder otros\carlos\spyder\spyder\widgets\findreplace.py", line 452, in set_editor
    self.editor.textChanged.disconnect(self.update_matches)
RuntimeError: wrapped C/C++ object of type CodeEditor has been deleted
```

Should that be solved in a different PR perhaps, maybe at #20221 ?

ccordoba12 commented:
Thanks @dalthviz for the review. I also encountered that problem and fixed it in PR #20221, specifically in commit [547348b](https://github.com/spyder-ide/spyder/pull/20221/commits/547348b962e3d677f9dc02fe71589da435aed20d) because it was catched by one of the new tests I added there.

CAM-Gerlach reviewed (commented):
LGTM overall, though could you maybe reduce the padding between the icon and the "clear field" (delete) button to about to half the current value? It looks rather strange currently and takes up more of the field's space than necessary. Thanks!

dalthviz reviewed (approved):
Thanks for the info @ccordoba12 , then this LGTM :+1:

I will leave this approved but I think that @CAM-Gerlach comment makes sense too, so if it is possible/not to complicated maybe it could be worthy to try to do the change before merging this

ccordoba12 commented:
> LGTM overall, though could you maybe reduce the padding between the icon and the "clear field" (delete) button to about to half the current value?

As it's almost the case, it was not trivial to implement in Qt, but I managed to do it. See my original description for the updated screenshot.

@dalthviz, could you check that it looks ok on Windows and Mac? Thanks!

CAM-Gerlach reviewed (approved):
LGTM, pending @dalthviz 's checks (I'm still on a trip right now, sorry)
